appservice/src/createAppService/SiteNameStep.ts

```diff
@@ -122,12 +127,104 @@ export class SiteNameStep extends AzureNameStep<SiteNameStepWizardContext> {
         return undefined;
     }
 
-    private async asyncValidateSiteName(client: WebSiteManagementClient, name: string): Promise<string | undefined> {
+    // For comprehensive breakdown of validation logic, please refer to: https://github.com/microsoft/vscode-azuretools/pull/1882#issue-2828801875
+    private async asyncValidateSiteName(context: SiteNameStepWizardContext, sdkClient: WebSiteManagementClient, name: string): Promise<string | undefined> {
+        name = name.trim();
+
+        let validationMessage: string | undefined;
+        if (!context.newSiteDomainNameLabelScope || context.newSiteDomainNameLabelScope === DomainNameLabelScope.Global) {
+            validationMessage ??= await this.asyncValidateGlobalCNA(sdkClient, name);
+        }
+
+        if (context.newSiteDomainNameLabelScope) {
+            validationMessage ??= await this.asyncValidateRegionalCNA(context, context.newSiteDomainNameLabelScope, name, context.resourceGroup?.name ?? context.newResourceGroupName);
+            validationMessage ??= await this.asyncValidateUniqueARMId(context, sdkClient, name, context.resourceGroup?.name ?? context.newResourceGroupName);
+        }
+
+        return validationMessage;
+    }
+
+    private async asyncValidateGlobalCNA(client: WebSiteManagementClient, name: string): Promise<string | undefined> {
         const nameAvailability: ResourceNameAvailability = await client.checkNameAvailability(name, 'Site');
         if (!nameAvailability.nameAvailable) {
             return nameAvailability.message;
         } else {
             return undefined;
         }
     }
+
+    private async asyncValidateRegionalCNA(context: SiteNameStepWizardContext, domainNameScope: DomainNameLabelScope, siteName: string, resourceGroupName?: string): Promise<string | undefined> {
+        if (!LocationListStep.hasLocation(context)) {
+            throw new Error(vscode.l10n.t('Internal Error: A location is required when validating a site name with regional CNA.'));
+        }
+        if (domainNameScope === DomainNameLabelScope.ResourceGroup && !resourceGroupName) {
```

Comment:
nit: Could probably be an else if since you only need one of these to throw an error.